docs/shared/features/maintain-typescript-monorepos.md

```diff
@@ -0,0 +1,286 @@
+# Maintain TypeScript Monorepos
+
+Keeping all the industry-standard tools involved in a large TypeScript monorepo correctly configured and working well together is a difficult task. And the more tools you add, the more opportunity there is for tools to conflict with each other in some way.
+
+In addition to [generating default configuration files](/features/generate-code) and [automatically updating dependencies](/features/automate-updating-dependencies) to versions that we know work together, Nx makes managing all the tools in your monorepo easier in two ways:
+
+- Rather than adding another tool that you have to configure, Nx configures itself to match the existing configuration of other tools.
+- Nx also enhances certain tools to be more usable in a monorepo context.
+
+## Auto-Configuration
+
+Whenever possible, Nx will detect the existing configuration settings of other tools and update itself to match.
+
+### Project Detection with Workspaces
+
+If your repository is using package manager workspaces, Nx will use those settings to find all the projects in your repository. So you don't need to define a project for your package manager and separately identify the project for Nx. The `workspaces` configuration on the left, allows Nx to detect the project graph on the right.
+
+{% side-by-side align="top" %}
+
+```json {% fileName="/package.json" %}
+{
+  "workspaces": ["apps/**", "packages/**"]
```

Comment:
Use `apps/*` and `packages/*`. We know some people run into problems with `**` picking up `package.json` it shouldn't, and the npm/pnpm/etc. docs all show `*` or explicit entries.